src/book/mod.rs

```diff
@@ -395,52 +396,49 @@ impl MDBook {
     }
 
     pub fn get_root(&self) -> &Path {
-        &self.root
+        self.config.get_root()
     }
 
-    pub fn set_dest(mut self, dest: &Path) -> Self {
-
-        // Handle absolute and relative paths
-        if dest.is_absolute() {
-            self.dest = dest.to_owned();
+    
+    pub fn with_destination<T: Into<PathBuf>>(mut self, destination: T) -> Self {
+        let root = self.config.get_root().to_owned();
+        if let Some(htmlconfig) = self.config.get_mut_html_config() {
+            htmlconfig.set_destination(&root, &destination.into());
         } else {
-            let dest = self.root.join(dest).to_owned();
-            self.dest = dest;
+            error!("There is no HTML renderer set...");
         }
 
         self
     }
+    
 
-    pub fn get_dest(&self) -> &Path {
-        &self.dest
-    }
-
-    pub fn set_src(mut self, src: &Path) -> Self {
-
-        // Handle absolute and relative paths
-        if src.is_absolute() {
-            self.src = src.to_owned();
-        } else {
-            let src = self.root.join(src).to_owned();
-            self.src = src;
+    pub fn get_destination(&self) -> Option<&Path> {
+        if let Some(htmlconfig) = self.config.get_html_config() {
+            return Some(htmlconfig.get_destination());
         }
 
+        None
+    }
+
+    pub fn with_source<T: Into<PathBuf>>(mut self, source: T) -> Self {
+        self.config.set_source(source);
         self
     }
 
-    pub fn get_src(&self) -> &Path {
-        &self.src
+    pub fn get_source(&self) -> &Path {
+        self.config.get_source()
     }
 
-    pub fn set_title(mut self, title: &str) -> Self {
-        self.title = title.to_owned();
+    pub fn with_title<T: Into<String>>(mut self, title: T) -> Self {
+        self.config.set_title(title);
         self
     }
 
     pub fn get_title(&self) -> &str {
-        &self.title
+        self.config.get_title()
     }
 
+/*
     pub fn set_author(mut self, author: &str) -> Self {
```

Comment:
get and set author?
No, we can remove them. I would like to remove the other methods as well and provide a better way to access the configuration, but they are too pervasively used throughout the code and a little to ambitious for this PR.